### ISSUE TITLE
Build failure with QT 5.10

### ISSUE THREAD

**Issue by LW-archlinux** [lbellonda/qxmledit#46]
`no matching function for call to ‘QPainter::drawStaticText(int, int, QString)’
         painter.drawStaticText(width() / 2, height() / 2, tr("Recalculating..."));`

[Full log](https://app.box.com/s/kzkvzr862hw78qvecui4iychbvcbdv5f)

Checking [1](http://code.qt.io/cgit/qt/qtbase.git/tree/dist/changes-5.10.0/?h=v5.10.0) and [2](http://doc.qt.io/qt-5/qpainter.html) it seems for qt 5.10 the function needs to be called like this:
`QPainter::drawStaticText(int, int, const QStaticText&)`

relevant snippet of log : 
```c++
visualization/datawidget.cpp: In member function ‘virtual void DataWidget::paintEvent(QPaintEvent*)’:
visualization/datawidget.cpp:339:81: error: no matching function for call to ‘QPainter::drawStaticText(int, int, QString)’
         painter.drawStaticText(width() / 2, height() / 2, tr("Recalculating..."));
                                                                                 ^
In file included from /usr/include/qt/QtGui/qpaintengine.h:47:0,
                 from /usr/include/qt/QtGui/QtGui:64,
                 from globals/includes/qtincludes.h:91,
                 from globals/includes/xmlEdit.h:77,
                 from precompiled_app.h:27:
/usr/include/qt/QtGui/qpainter.h:409:10: note: candidate: void QPainter::drawStaticText(const QPointF&, const QStaticText&)
     void drawStaticText(const QPointF &topLeftPosition, const QStaticText &staticText);
          ^~~~~~~~~~~~~~
/usr/include/qt/QtGui/qpainter.h:409:10: note:   candidate expects 2 arguments, 3 provided
/usr/include/qt/QtGui/qpainter.h:865:13: note: candidate: void QPainter::drawStaticText(const QPoint&, const QStaticText&)
 inline void QPainter::drawStaticText(const QPoint &p, const QStaticText &staticText)
             ^~~~~~~~
/usr/include/qt/QtGui/qpainter.h:865:13: note:   candidate expects 2 arguments, 3 provided
/usr/include/qt/QtGui/qpainter.h:870:13: note: candidate: void QPainter::drawStaticText(int, int, const QStaticText&)
 inline void QPainter::drawStaticText(int x, int y, const QStaticText &staticText)
             ^~~~~~~~
/usr/include/qt/QtGui/qpainter.h:870:13: note:   no known conversion for argument 3 from ‘QString’ to ‘const QStaticText&’
```

**Comment by lbellonda** [comment]
Hello, compatibility with Qt5.10 has not been verified.
I will check asap.
Are you using some distribution with Qt5.10?

**Comment by LW-archlinux** [comment]
yes, i'm running Arch Linux .
AL follows a rolling release model aiming for latest stable versions.
QT 5.10 entered our repos around 13 december 2017.

While qxmledit is not in arch repos, I maintain an AUR package for your great editor that allows AL users to build it from source themselves . ( see https://aur.archlinux.org/packages/qxmledit ).

(The binary 0.9.8 I built against qt 5.9.3 still works fine ) .

**Comment by lbellonda** [comment]
Hello, I opened a branch r-0.9.9.2 to address the issue. Can you test it?

Thanks.

**Comment by LW-archlinux** [comment]
the new branch builds and installs fine.
I opened 2 xml files, edited a few fields and didn't notice anything strange.

**Comment by lbellonda** [comment]
An official patch will be released in approx one or two weeks (0.9.9.2).

**Comment by lbellonda** [comment]
Hello, version 0.9.9.2 has been released today.
Closing this ticket.
